docs/joss/paper.md

```diff
@@ -0,0 +1,203 @@
+and research playground with an (almost) everything-flexible attitude.
+It can be thought of as a conceptual reinvention of the Fortran SPEEDY model [@Molteni2003]
+in the Julia programming language [@Bezanson2017].
+
+SpeedyWeather.jl internally uses three sub-modules `SpeedyTransforms`, `RingGrids`, and
+`LowerTriangularMatrices` to perform spherical harmonic transforms and interpolations
+between various grids and the spectral space. `RingGrids` discretize the sphere
+on iso-latitude rings and the spectral space is defined by the `LowerTriangularMatrices`
+of the spherical harmonic coefficients. These three modules are independently usable
+and therefore make SpeedyWeather.jl, beyond its main purpose of simulating the weather,
+also a library for the analysis of gridded data on the sphere.
+Running and analysing simulations can be interactively combined, enhancing user
+experience and productivity.
+
+The user interface of SpeedyWeather.jl is heavily influenced by
+the Julia ocean model Oceananigans.jl [@Ramadhan2020].
+A monolithic interface based on parameter files is avoided in favor of a
+library-style interface in which users write code to run models rather than
+merely supplying parameters and input arrays.
+A model is created bottom-up by first defining the discretization
+and any non-default model components with its respective parameters.
+All components are then collected into a single model object, which, once
+initialized, returns a simulation object that contains the entire model state,
+work arrays and parameters, that can be run, analysed or changed.
+While these steps can be written into a script for reproducibility,
+the same steps can be executed and interacted with one-by-one in
+Julia's read-evaluate-print loop (REPL). We thereby reach an interactivity
+far beyond a monolithic interface that is limited to the options provided.
+At the same time, sensible default arguments enable even inexperienced users
+to run simulations in just a few lines of code. 
+
+To be extensible and composable with new
+model components, SpeedyWeather.jl relies on Julia's multiple dispatch
+programming paradigm [@Bezanson2017]. Every model component
+is defined as a new type. For example, to define a new way how to calculate
+the precipitation due to the physical process of large-scale condensation,
+one would define `MyCondensation` as a new subtype of `AbstractCondensation`.
+One then only needs to extend the `initialize!` and `condensation!`
+functions for this new type. Passing on `condensation = MyCondensation()`
+to the model constructor then implements this new model component without
+the need to branch off or overwrite existing model components.
+Conceptually similar scientific modelling paradigms have been very successful
+in the Python-based generic partial differential equation solver Dedalus [@Burns2020]
+and the Julia ocean model Oceananigans.jl [@Ramadhan2020].
```

Comment:
Similar to my comments from above. Please justify why there is a need to reword the entire paragraph, especially because you introduced several points that are less correct, less concise as they were before. You write "distinct type", which I'd like to avoid because these types are sub and supertypes of each other, so they have a relation and depending on the type tree aren't always distinguished and can be subtyped. You write "functions need modification for this type" no, that's not how multiple dispatch works. There's a reason I wrote "extend" rather than "modify" because with multiple dispatch you don't have to modify or overwrite the previous definition of a function, rather you extend it by defining a new method, that's very different and in fact our entire approach wouldn't work if it wasn't for that!